packages/compass-indexes/src/components/indexes/indexes.tsx

```diff
@@ -123,23 +144,28 @@ const mapState = ({
   description,
   serverVersion,
   appRegistry,
-  regularIndexes: { indexes, isRefreshing, error },
-  searchIndexes: { status },
+  regularIndexes,
+  searchIndexes,
 }: RootState) => ({
-  indexes,
+  indexes: regularIndexes.indexes,
+  searchIndexes: searchIndexes.searchIndexes,
   isWritable,
   isReadonlyView,
   description,
-  error,
+  regularError: regularIndexes.error,
+  searchError: searchIndexes.error,
   localAppRegistry: (appRegistry as any).localAppRegistry,
-  isRefreshing,
+  isRefreshing:
+    regularIndexes.isRefreshing || searchIndexes.status === 'REFRESHING',
```

Comment:
We now have two variables (one for regular indexes and one for search indexes) that mean "refreshing". Which is just weird. But then the toolbar needs to know whether you're already refreshing. Will refreshing the regular indexes disable refreshing the search indexes? The design is a bit weird IMHO.

Reply:
In fact if searchIndexes.status === 'PENDING' then that probably also counts as loading..

Reply:
Currently in indexes we do not have initial loading, but we can introduce that for search indexes (status === PENDING). 

These all props are tab based - regular/search and having them that way will help us avoid `regularIndexes.isRefreshing || searchIndexes.status === 'REFRESHING'` and show correct state per tab. Similarly for error. 

We can also connect these components (toolbar, search indexes table and regular indexes table) individually to redux and that can clean it up here a lot.